op-node/rollup/derive/channel_out.go

```diff
@@ -136,6 +173,70 @@ func (co *ChannelOut) AddBatch(batch *BatchData) (uint64, error) {
 	return uint64(written), err
 }
 
+// writeSpanBatch appends a SingularBatch to the channel's SpanBatch.
+// A channel can have only one SpanBatch. And compressed results should not be accessible until the channel is closed, since the prefix and payload can be changed.
+// So it resets channel contents and rewrites the entire SpanBatch each time, and compressed results are copied to reader after the channel is closed.
```

Comment:
How does this integrate with the shadow/non-shadow compressor? I don't love the default behavior of recompressing the channel every time we add a new block, but the shadow compressor does do something similar.

Reply:
I agree that recompressing the entire channel every time may not be that great. But we couldn't think of a better alternative for that part due to the data layout of the span-batch.

As you said, using the shadow compressor makes unnecessary compression. so the ratio compressor will be more suitable for span-batch.

Reply:
Low estimation accuracy of the ratio compressor is not a problem, because `ChannelOut` does not use the compressor's estimation logic when it's working with span-batch.